src/components/Menu/Menu.tsx

```diff
@@ -251,10 +261,19 @@ export function ButtonMenu({menuProps, wrapperProps, ...buttonProps}: ButtonMenu
 
   return (
     <Box cs={wrapperProps}>
-      <Button ref={buttonRef} onClick={() => setShown(!shown)} {...buttonProps} />
-      {shown && (
-        <Menu {...menuProps} onClose={() => setShown(false)} anchorEl={buttonRef} />
-      )}
+      <ClickAwayListener onClickAway={() => setShown(false)}>
```

Comment:
Question; why did we move this here instead of using the click away listener in Menu?

Reply:
This was brought up in design QA where what was expected was if we click on button in button menu, it should open up the menu, then when clicking again of button, it should close the menu. Previous implementation was when we click on button it keeps on opening because it is also considered clickaway from menu, thus changing the shown prop

Reply:
Okay, I think the solution there would be to prevent event propogation on the button onClick, instead of moving the click away listener around and introducing additional props complexity?